src/deltalake_ext.rs

```diff
@@ -112,24 +127,39 @@ impl DeltaParquetWriter {
 
         Ok(Self {
             table_path,
+            table,
             storage,
             arrow_schema_ref,
             writer_properties,
             cursor,
             arrow_writer,
+            partition_columns,
             partition_values: HashMap::new(),
             num_records: 0,
         })
     }
 
+    pub async fn last_transaction_version(
+        &mut self,
+        app_id: &str,
+    ) -> Result<Option<DeltaDataTypeVersion>, DeltaWriterError> {
```

Comment:
I'm curious why bother wrapping a `Result` around an `Option`? Is there a case where a `None` would be returned that wouldn't constitute an error worth propagating?

Reply:
In this case - yes, there will not be a txn entry for kafka-delta-ingest until after the first write. An `Ok` indicates we have read the log successfully. If the caller is the first to write a txn entry the returned value will be a `None`.